grudge/trace_pair.py

```diff
@@ -318,8 +319,20 @@ def interior_trace_pair(dcoll: DiscretizationCollection, vec) -> TracePair:
     return local_interior_trace_pair(dcoll, vec)
 
 
+@dataclass(frozen=True)  # for KeyBuilder support
+class CommTag:
```

Comment:
It doesn't really make sense for all `CommTag`s to be subclasses of this. Some may have data contained in them.

Reply:
I added a small test in [9a63df4](https://github.com/inducer/grudge/pull/319/commits/9a63df4e774e517918ae944dbc89e8b2bac17568) with subclassed dataclasses, is that what you had in mind?

Reply:
Ok, would you like to keep the `CommTag` class as an optional way to construct the comm tags? i removed the warning in 945406730bd968fdde9688dbb7ce604a329b7110.